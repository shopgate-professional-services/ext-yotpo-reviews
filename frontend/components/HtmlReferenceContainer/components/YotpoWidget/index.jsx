import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './style';
import getYotpo from '../../../../helpers/getYotpo';

/**
 * Mounts yotpo script.
 */
class YotpoScript extends Component {
  static propTypes = {
    config: PropTypes.shape().isRequired,
    productId: PropTypes.string.isRequired,
    product: PropTypes.shape(),
  };

  static defaultProps = {
    product: null,
  }

  /**
   * Constructs
   * @param {Objects} props props
   */
  constructor(props) {
    super(props);
    this.yotpoMountTimeout = undefined;
  }

  /**
   * Mounts Yotpo script.
   */
  componentDidMount() {
    if (!window.yotpo) {
      getYotpo(this.props.config.yotpoAppKey);
      return;
    }
    window.yotpo.initialized = false;
    window.yotpo.clean();
    this.yotpoMountTimeout = setTimeout(() => {
      window.yotpo.initWidgets();
      this.yotpoMountTimeout = undefined;
    }, 500);
  }

  /**
   * Cleanup timeout request
   */
  componentWillUnmount() {
    clearTimeout(this.yotpoMountTimeout);
  }

  /**
   * Renders.
   * @return {JSX}
   */
  render() {
    if (!this.props.product) {
      return null;
    }
    return (
      <div>
        <div
          className={`yotpo bottomLine ${styles.padding}`}
          data-appkey={`${this.props.config.yotpoAppKey}`}
          data-domain=""
          data-product-id={`${this.props.productId}`}
          data-name={`${this.props.product.name}`}
          data-url={`${this.props.product.productUrl}`}
          data-image-url={`${this.props.product.featuredImageUrl}`}
          data-description=""
        />
        <div
          className={`yotpo yotpo-main-widget ${styles.container} ${styles.padding}`}
          data-product-id={`${this.props.productId}`}
          data-name={`${this.props.product.name}`}
          data-url={`${this.props.product.productUrl}`}
          data-image-url={`${this.props.product.featuredImageUrl}`}
          data-description=""
        />
      </div>
    );
  }
}

export default YotpoScript;
