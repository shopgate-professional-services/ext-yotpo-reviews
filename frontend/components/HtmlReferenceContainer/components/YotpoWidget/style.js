import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables } = themeConfig;

const padding = css({
  padding: `0px ${variables.gap.big}px 0px ${variables.gap.big}px`,
}).toString();

const container = css({
  maxWidth: '100%',
  wordWrap: 'break-word',
  wordBreak: 'bread-word',
  ' .socialize-wrapper': {
    display: 'none !important',
  },
}).toString();

export default {
  container,
  padding,
};
