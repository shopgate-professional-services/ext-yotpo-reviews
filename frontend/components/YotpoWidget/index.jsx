import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './style';
import getYotpo from '../../helpers/getYotpo';

/**
 * Mounts yotpo script.
 */
class YotpoScript extends Component {
  static propTypes = {
    config: PropTypes.shape().isRequired,
    productId: PropTypes.string.isRequired,
    product: PropTypes.shape(),
  };

  static defaultProps = {
    product: null,
  }

  /**
   * Mounts Yotpo script.
   */
  componentDidMount() {
    if (!window.yotpo) {
      getYotpo(this.props.config.yotpoAppKey);
      return;
    }
    window.yotpo.initialized = false;
    window.yotpo.clean();
    setTimeout(() => { window.yotpo.initWidgets(); }, 500);
  }

  /**
   * Update for yotpo fetching for different variant
   * @param {Object} prevProps prevProps
   */
  componentDidUpdate(prevProps) {
    if (this.props.productId !== prevProps.productId) {
      window.yotpo.initialized = false;
      window.yotpo.clean();
      // 1500 Additional ms to account for refresh of widget properties and fetching of new widget.
      setTimeout(() => {
        window.yotpo.refreshWidgets();
        window.yotpo.initWidgets();
      }, 2000);
    }
  }

  /**
   * Renders.
   * @return {JSX}
   */
  render() {
    if (!this.props.product) {
      return null;
    }
    return (
      <div>
        <div
          className={`yotpo bottomLine ${styles.padding}`}
          data-appkey={`${this.props.config.yotpoAppKey}`}
          data-domain=""
          data-product-id={`${this.props.productId}`}
          data-name={`${this.props.product.name}`}
          data-url={`${this.props.product.productUrl}`}
          data-image-url={`${this.props.product.featuredImageUrl}`}
          data-description=""
        />
        <div
          className={`yotpo yotpo-main-widget ${styles.container} ${styles.padding}`}
          data-product-id={`${this.props.productId}`}
          data-name={`${this.props.product.name}`}
          data-url={`${this.props.product.productUrl}`}
          data-image-url={`${this.props.product.featuredImageUrl}`}
          data-description=""
        />
      </div>
    );
  }
}

export default YotpoScript;
