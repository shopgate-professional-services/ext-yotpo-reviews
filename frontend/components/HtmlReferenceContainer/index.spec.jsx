import React from 'react';
import { mount } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import YotpoWidget from './components/YotpoWidget';
import HtmlReferenceContainer from './index';

jest.mock('./components/YotpoWidget', () => () => (<div>YotpoWidget</div>));

const store = createMockStore();

describe('HtmlReferenceContainer', () => {
  it('should render HtmlReferenceContainer', () => {
    const component = mount((
      <Provider store={store}>
        <HtmlReferenceContainer
          config={{ yotpoAppKey: 'mock' }}
          productId="product_one"
        />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
    expect(component.find(YotpoWidget).length).toBe(1);
  });
});
