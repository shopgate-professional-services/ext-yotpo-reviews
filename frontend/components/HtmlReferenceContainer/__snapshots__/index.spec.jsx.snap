// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlReferenceContainer should render HtmlReferenceContainer 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HtmlReferenceContainer
    config={Object {}}
    openPageExtern={[Function]}
    product={Object {}}
    productId="product_one"
  >
    <div>
      <Component
        config={Object {}}
        product={Object {}}
        productId="product_one"
      >
        <div>
          YotpoWidget
        </div>
      </Component>
    </div>
  </HtmlReferenceContainer>
</Provider>
`;
